astropy/coordinates/sites.py

```diff
@@ -0,0 +1,115 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Observatories accessible without internet access originate from the IRAF
+Observatory Database, and are stored in ``data/observatories.json``.  This is
+inteded mainly as a fallback file, and the online file is where new changes
+should go.
+
+Additions or corrections to the observatory list can be submitted via Pull
+Request to the [astropy-data GitHub repository](https://github.com/astropy/astropy-data),
+updating the ``location.json`` file.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import json
+from difflib import get_close_matches
+
+from ..utils.data import get_pkg_data_contents, get_file_contents
+from .earth import EarthLocation
+from .. import units as u
+
+
+class SiteRegistry(object):
+    """
+    A bare-bones registry of EarthLocation objects.
+    """
+    def __init__(self):
+        # the keys to this are always lower-case
+        self._site_dict = {}
```

Comment:
I'm starting to nit-pick, but since this is already a site registry, I'd call these `_locations` and `_names` (certainly no need to append `_dict` -- it *is* a dict, after all!).